[FX] Validate data type of target on Node Construction

yqhu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64050

Differential Revision: [D30585535](https://our.internmc.facebook.com/intern/diff/D30585535)

yqhu committed:
[FX] Validate data type of target on Node Construction

[ghstack-poisoned]

jamesr66a reviewed (commented):
Great, thanks! Just a comment about our error reporting discipline, I would really appreciate if you could improve how we handle the errors here. Also, please add a commit message describing what this change is doing. Also if you say "Closes https://github.com/pytorch/pytorch/issues/63490", GitHub will automatically close the issue when your PR is merges

yqhu committed:
Update on "[FX] Validate data type of target on Node Construction"


Differential Revision: [D30585535](https://our.internmc.facebook.com/intern/diff/D30585535)

[ghstack-poisoned]

jamesr66a reviewed (approved):
![image](https://user-images.githubusercontent.com/4685384/131164203-72908894-2401-4258-a76a-b240e162a0fb.png)
